core/docs/d7yxc-operator_manual.md

```diff
@@ -1,21 +1,274 @@
+- Managing secrets available to pipelines in the session
+- Managing the resolution of unpinned sources to specific versions
+  - E.g. the mapping of `image:latest` to a specific digest, mapping of a git branch to a specific commit, etc.
+  - This mapping will happen once per-session on first use and be frozen for the rest of the session after
+
+Sessions are expected to be run on the same host as the SDK client; otherwise local resources like directories and sockets will differ from that which is local to the client.
+
+### Runner
+
+A runner is the "backend" of sessions where containers are actually executed.
+
+Runners are responsible for:
+
+- Executing containers specified by pipelines
+- Pulling container images, git repos and other sources needed for pipeline execution
+- Pushing container images to registries
+- Managing the cache backing pipeline execution
+
+The runner is distributed as a container image, making it easy to run on various container runtimes like Docker, Kubernetes, Podman, etc.
+
+It's typically run persistently, as opposed to sessions which only last for the duration of `dagger run` or `dagger.Connect` in an SDK.
+
+## FAQ
+
+### What are the steps for using a custom provisioned runner?
+
+The details are [here](#runner-details), but the consolidated steps are:
+
+1. Determine the runner version required by checking the release notes of the SDK you intend to use
+1. If changes to the base image are needed, make those and push them somewhere. If no changes are needed, just use it as is.
+1. Start the runner image in your target of choice, keeping the [requirements](#execution-requirements) and [configuration](#configuration) in mind.
+1. Export the `_EXPERIMENTAL_DAGGER_RUNNER_HOST` environment variable with [a value pointing to your target](#connection-interface).
+1. Call `dagger run` or execute SDK code directly with that environment variable set.
+
+### What compatibility is there between SDK, CLI and Runner versions?
+
+This is only needed if you are using a custom provisioned runner or a pre-installed CLI. If you are just using an SDK directly a CLI and runner will be provisioned automatically at compatible versions.
+
+The CLI+Runner share a version number. SDKs have their own version number, but they are currently only guaranteed to be compatible with a single CLI+Runner version at at time.
+
+To ensure that an SDK will be used with a compatible CLI and runner:
+
+1. Check the release notes of the SDK, which will point to the required CLI+Runner version
+1. If using a custom provisioned runner, use the base image at that version as detailed [here](#distribution-and-versioning).
+1. If using a pre-installed CLI, install the CLI at that version as detailed [here](#versioning)
+1. Once the runner and/or CLI are setup, you are safe to upgrade your SDK to the newest version.
+
+## Runner Details
+
+### Distribution and Versioning
+
+The runner is distributed as a container image at `ghcr.io/dagger/engine`.
```

Comment:
Confused by the mix of the terms "runner" and "engine".

Reply:
Yeah I agree, I think we settled on the term runner at some point, but unfortunately after we already created the image repo where the image is called engine.

I guess maybe the terminology could be "clarified" such that engine refers to the session helper bin + runner (aka buildkitd for now)? But I guess that still wouldn't make sense because we don't distribute the session binary as part of the engine image anymore.

And given our work with gha runners, use of the term runner here might start to get really confusing. So maybe we can just drop "runner" from here entirely and settle on engine instead?

cc @shykes don't want to block merging on this terminology issue but is something that needs to be addressed, let me know if you have thoughts.